force-app/main/default/classes/ListDonation.cls

```diff
@@ -0,0 +1,111 @@
+        List<DonationHistory> partialSoftCredit = verifyPartialSoftCreditConsition(donationsId, donations);
+        donationHistory.addAll(partialSoftCredit);
+        System.debug(donationHistory);
+        return donationHistory;
+    }
+    
+    public static List<DonationHistory> verifyHardCreditCondition(List<OpportunityContactRole> donations){
+        List<DonationHistory> hardCreditList = new List<DonationHistory>();
+        for(OpportunityContactRole opp: donations){
+            if (opp.isPrimary) {
+                DonationHistory hardCredit = new DonationHistory();
+                hardCredit.id = opp.Id;
+                hardCredit.name = opp.Opportunity.Name;
+                hardCredit.closeDate = opp.Opportunity.CloseDate;
+                hardCredit.role = opp.Role;
+                hardCredit.amount = opp.Opportunity.Amount;
+                hardCredit.creditType = HARD_CREDIT;
+                hardCreditList.add(hardCredit);
+            }
+        }
+        return hardCreditList;
+    }
+    
+    public static List<DonationHistory> verifySoftCreditCondition(List<OpportunityContactRole> donations){
+        //with the list of donations verify if the Role match the filter rule value on the NPSP_ClosedWon_Opps_SC filter group
+        Filter_Rule__mdt roles = [SELECT Constant__c FROM Filter_Rule__mdt WHERE QualifiedApiName = 'NPSP_OppsSoftCreditRoles'];
+        List<String> softCreditRoles = roles.Constant__c.split(';');
+        List<DonationHistory> softCreditList = new List<DonationHistory>();
+        for(OpportunityContactRole opp: donations){
+            if (softCreditRoles.contains(opp.Role)) {
+                DonationHistory softCredit = new DonationHistory();
+                softCredit.id = opp.Id;
+                softCredit.name = opp.Opportunity.Name;
+                softCredit.closeDate = opp.Opportunity.CloseDate;
+                softCredit.role = opp.Role;
+                softCredit.amount = opp.Opportunity.Amount;
+                softCredit.creditType = HARD_CREDIT;
+                softCreditList.add(softCredit);
+            }
+        }
+        return softCreditList;
+    }
+
+    public static List<Id> filterVerifiedIds(List<DonationHistory> donationHistory, List<OpportunityContactRole> donations){
+        List<Id> donationsId = new List<Id>();
+        List<Id> donationHistoryList = new List<Id>();
+        //Set<String> donationHistorySet = (new Map<String, DonationHistory>(donationHistory)).keySet();
+        for(DonationHistory dh: donationHistory){
+            donationHistoryList.add(dh.id);
+        }
+        for(OpportunityContactRole opp: donations){
+            if (!donationHistoryList.contains(opp.id)) {
+                donationsId.add(opp.Id);
+            }
+        }
+        return donationsId;
+    }
+
+    public static List<DonationHistory> verifyPartialSoftCreditConsition(List<Id> donationsId, List<OpportunityContactRole> donations){
```

Comment:
typo, method should be named _verifyPartialSoftCreditCondition_

Reply:
fixed